### PR TITLE
User levels for few endpoints

### PR DESCRIPTION
**Pull request by lzganec**
PUT in VolunteerRequest is kinda useless right now since we don't have any fields that make sense to be updated, but I left it in case something changes.

### CODE REVIEW

File: server/tests/api/admin/UsersTest.php [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    // {
+    //     $response = $this->http->request('GET', '/api/users.php?number_per_page=11');
+    //     $this->assertEquals(200, $response->getStatusCode());
+    //     $contentType = $response->getHeaders()["Content-Type"][0];
+    //     $this->assertEquals("application/json", $contentType);
+    //     $data = json_decode($response->getBody(true), true);
+    //     $this->assertEquals(11, count($data));
+    // }
+    
+    // TODO: Admins should be able to create admins
```
Comment (BLamy) [inline server/tests/api/admin/UsersTest.php]: - [ ] Admins should be able to create admins

File: server/tests/api/admin/UsersTest.php [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    //     $this->assertEquals('Admin', $passenger->lastName);
+    //     $this->assertEquals('admin', $passenger->role);
+    //     $this->assertEquals('admin', $passenger->role);
+    //     $this->assertEquals('2036259401', $passenger->phone);
+    //     $this->assertEquals('testAdmin@websterwasps.com', $passenger->email);
+    //     $this->assertEquals(1, $passenger->wantsSMS);
+    //     $this->assertEquals(1, $passenger->wantsEmail);
+    // }
+
+    // TODO: Admins should be able to create dispatchers
```
Comment (BLamy) [inline server/tests/api/admin/UsersTest.php]: - [ ] Admins should be able to create dispatchers

File: server/tests/api/admin/UsersTest.php [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+    //     $this->assertEquals('Test', $passenger->firstName);
+    //     $this->assertEquals('Dispatcher', $passenger->lastName);
+    //     $this->assertEquals('dispatcher', $passenger->role);
+    //     $this->assertEquals('2034229403', $passenger->phone);
+    //     $this->assertEquals('testDispatcher@websterwasps.com', $passenger->email);
+    //     $this->assertEquals(1, $passenger->wantsSMS);
+    //     $this->assertEquals(1, $passenger->wantsEmail);
+    // }
+
+    // TODO: Admins should be able to create Drivers
```
Comment (BLamy) [inline server/tests/api/admin/UsersTest.php]: - [ ] Admins should be able to create drivers

File: server/tests/api/admin/UsersTest.php [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+                'firstName' => 'Test',
+                'lastName' => 'Driver',
+                'password' => 'password',
+                'phone' => '2034259481'
+            ]
+        ]);
+        $this->assertEquals(500, $response->getStatusCode());
+    }
+
+    // TODO: There seem to be issues with updating users.
```
Comment (BLamy) [inline server/tests/api/admin/UsersTest.php]: - [ ]  There seem to be issues with updating users.

**Comment by lzganec** [comment]
@BLamy From the short skim through the tests, the problem is that you cannot create any user with a specified role. The way it's set up right now is that all users created are passengers by default and the admin can then update it through the PUT method. I can make it so that admin accessing the POST for users can create it if updating is inconvenient. Let me know which way you'd prefer.

As for the updating the user, the ID is only accepted through the body, not the URL. I think it got messed up between you setting some IDs to be fetched from the URL and me setting some through the body. Will need to go through all methods and make it available to pass ID through URL as I had something messed up in my head when I was putting everything through body.

**Comment by BLamy** [comment]
I'm going to merge this now just so other people don't get out of sync with us. You can create another one and then we will work off of that.

**Pull request merged**